network/tls_wrapper.go

```diff
@@ -49,19 +49,39 @@ type TLSConnectionWrapper struct {
 var ErrEmptyTLSConfig = errors.New("empty TLS config")
 
 var (
-	tlsCA         string
-	tlsKey        string
-	tlsCert       string
-	tlsAuthType   int
-	tlsServerName string
+	tlsCA                  string
+	tlsKey                 string
+	tlsCert                string
+	tlsAuthType            int
+	tlsServerName          string
+	tlsOcspURL             string
+	tlsOcspClientURL       string
+	tlsOcspDbURL           string
+	tlsOcspRequired        string
+	tlsOcspFromCert        string
+	tlsOcspCheckWholeChain bool
+	tlsCrlURL              string
+	tlsCrlFromCert         string
+	tlsCrlCacheSize        int
+	tlsCrlCacheTime        int
 )
 
-// RegisterTLSBaseArgs register CLI args tls_ca|tls_key|tls_cert|tls_auth which allow to get tls.Config by NewTLSConfigFromBaseArgs function
+// RegisterTLSBaseArgs register CLI args tls_ca|tls_key|tls_cert|tls_auth|tls_ocsp_url|tls_ocsp_client_url|tls_ocsp_db_url|tls_ocsp_required|tls_ocsp_from_cert|tls_crl_url|tls_crl_from_cert which allow to get tls.Config by NewTLSConfigFromBaseArgs function
 func RegisterTLSBaseArgs() {
 	flag.StringVar(&tlsCA, "tls_ca", "", "Path to root certificate which will be used with system root certificates to validate peer's certificate")
 	flag.StringVar(&tlsKey, "tls_key", "", "Path to private key that will be used for TLS connections")
 	flag.StringVar(&tlsCert, "tls_cert", "", "Path to certificate")
 	flag.IntVar(&tlsAuthType, "tls_auth", int(tls.RequireAndVerifyClientCert), "Set authentication mode that will be used in TLS connection. Values in range 0-4 that set auth type (https://golang.org/pkg/crypto/tls/#ClientAuthType). Default is tls.RequireAndVerifyClientCert")
+	flag.StringVar(&tlsOcspURL, "tls_ocsp_url", "", "OCSP service URL")
+	flag.StringVar(&tlsOcspClientURL, "tls_ocsp_client_url", "", "OCSP service URL, for client certificates only")
+	flag.StringVar(&tlsOcspDbURL, "tls_ocsp_database_url", "", "OCSP service URL, for database certificates only")
+	flag.StringVar(&tlsOcspRequired, "tls_ocsp_required", "yes", "Whether we need OCSP response in order to accept certificate")
+	flag.StringVar(&tlsOcspFromCert, "tls_ocsp_from_cert", "prefer", "How should we threat OCSP server described in certificate itself")
+	flag.BoolVar(&tlsOcspCheckWholeChain, "tls_ocsp_check_whole_chain", false, "Whether to check whole certificate chain, false = only end certificate")
+	flag.StringVar(&tlsCrlURL, "tls_crl_url", "", "CRL URL")
+	flag.StringVar(&tlsCrlFromCert, "tls_crl_from_cert", "use", "How should we treat CRL URL described in certificate itself")
+	flag.IntVar(&tlsCrlCacheSize, "tls_crl_cache_size", 16, "Size of in-memory LRU cache for storing fetched CRLs, 0 = unlimited")
+	flag.IntVar(&tlsCrlCacheTime, "tls_crl_cache_time", 60, "How long cached CRL is considerend valid, seconds, max = 300")
```

Comment:
This is strangely familiar to the list from `acra-server` and friends 🤔  🤔  🤔

Reply:
This function `RegisterTLSBaseArgs()` (as well as `NewTLSConfigFromBaseArgs()`) is used in Acra EE, in acra-translator.
It makes sense there, no need to extend acra-translator's `main()`.
In Acra CE server & connector these functions are not called at all, so these two apps create TLS config in a different way.